lib/iris/tests/unit/util/test_equalise_cubes.py

```diff
@@ -0,0 +1,266 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the BSD license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Unit tests for the :func:`iris.util.equalise_cubes` function."""
+
+from cf_units import Unit
+import numpy as np
+from numpy.random import Generator
+import pytest
+
+from iris.coords import DimCoord
+from iris.cube import Cube
+from iris.util import equalise_cubes
+
+
+def _scramble(inputs, rng=95297):
+    # Reorder items to check that order does not affect the operation
+    if not isinstance(rng, Generator):
+        rng = np.random.default_rng(rng)
+    n_inputs = len(inputs)
+    # NOTE: make object array of explicit shape + fill it,
+    # since np.array(inputs) *fails* specifically with a list of metadata objects
+    inputs_array = np.empty((n_inputs,), dtype=object)
+    inputs_array[:] = inputs
+    n_inputs = inputs_array.shape[0]
+    scramble_inds = rng.permutation(n_inputs)
+    inputs_array = inputs_array[scramble_inds]
+    # Modify input list **BUT N.B. IN PLACE**
+    inputs[0:] = inputs_array
```

Comment:
Could `[0:]` not be replaced by `[:]`?

Reply:
Yes it can (!) : https://github.com/SciTools/iris/pull/6257/commits/ae992ffd333afc9493627916ba5564d73b24d953

I had simply misunderstood how this works